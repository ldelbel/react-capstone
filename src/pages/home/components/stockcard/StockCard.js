import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { useHistory } from 'react-router-dom';
import styles from './StockCard.module.scss';

const StyledDiv1 = styled.div`
  width: 31%;
  min-width: 250px;
  height: 200px;
  background: #ffffff;
  border-radius: 10px;
  margin-right: 15px;
  margin-top: 15px;
`;

const StyledDiv2 = styled.div`
height: 45%;
background: #ffffff;
padding: 0;
width: 90%;
border-bottom: 1px solid rgba(33,37,41,0.12);
`;
const StyledDiv3 = styled.div`
width: 55px;
height: 55px;
border-radius: 50%;
background-size: contain, auto;
background-position: center;
background-repeat: no-repeat;
`;

const StyledDiv4 = styled.div`
background: rgba(0,123,255,0.43);
color: var(--blue);
font-weight: bold;
padding: 0;
padding-right: 3px;
padding-left: 3px;
font-size: 12px;
`;

const StyledDiv5 = styled.div`
background: rgba(220,53,69,0.44);
color: var(--danger);
font-weight: bold;
padding-right: 3px;
padding-left: 3px;
font-size: 12px;
`;
const StyledDiv6 = styled.div`
color: #212529;
background: var(--dark);
height: 25%;
border-bottom-right-radius: 9px;
border-bottom-left-radius: 9px;
`;

const StyledBtn1 = styled.button`
color: var(--light);
background: var(--success);
font-family: 'Alfa Slab One', cursive;
border-style: none;
border-bottom-style: none;

&:hover {
  background: var(--light);
  color: var(--success);
}
`;

const StockCard = props => {
  const { profile } = props;
  const history = useHistory();
  // const [img, setImg] = useState(false);

  const handleClick = () => {
    history.push(`/details/${profile.symbol}`);
  };

  // useEffect(() => {
  //   setImg(true);
  // }, [profile]);

  return (
    <StyledDiv1>
      <div className="d-flex flex-column justify-content-start align-items-center h-75">
        <StyledDiv2 className="d-flex flex-row align-items-center">
          <div className="d-flex align-items-center w-50 h-100">
            <StyledDiv3 style={{ backgroundImage: `url(https://financialmodelingprep.com/image-stock/${profile.symbol}.jpg)` }} />
          </div>
          <div className="d-flex flex-column justify-content-start align-items-end w-50 h-100">
            <span style={{
              fontFamily: 'Alfa Slab One, cursive', fontSize: '25px', color: 'var(--dark)', height: '30px',
            }}
            >
              {profile.symbol}
            </span>
            <span style={{ color: 'var(--blue)', fontFamily: 'Alfa Slab One, cursive', fontSize: '10px' }}>{profile.exchangeShortName}</span>
          </div>
        </StyledDiv2>
        <div className={`w-100 p-0 ${styles.h55}`}>
          <div className={`d-flex align-items-end ${styles.h40pl10}`}>
            <span className={`text-nowrap text-truncate flex-nowrap ${styles.ffafs18}`}>
              {profile.companyName}
            </span>
          </div>
          <div className={`d-flex flex-column justify-content-center align-items-end ${styles.h60fs13}`}>
            <StyledDiv4><span>{profile.sector}</span></StyledDiv4>
            <StyledDiv5><span>{profile.industry}</span></StyledDiv5>
          </div>
        </div>
      </div>
      <StyledDiv6 className="d-flex flex-row justify-content-around align-items-center">
        <StyledBtn1 className="btn btn-primary" type="button" onClick={handleClick}>Details</StyledBtn1>
        <span className={styles.ffacyfs23}>{`U$ ${profile.price.toFixed(2)}`}</span>
      </StyledDiv6>
    </StyledDiv1>
  );
};

StockCard.propTypes = {
  profile: PropTypes.shape({
    symbol: PropTypes.string.isRequired,
    companyName: PropTypes.string.isRequired,
    exchangeShortName: PropTypes.string.isRequired,
    sector: PropTypes.string.isRequired,
    industry: PropTypes.string.isRequired,
    price: PropTypes.number.isRequired,
  }).isRequired,
};

export default StockCard;
