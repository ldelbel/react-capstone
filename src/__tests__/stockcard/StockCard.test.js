import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import StockCard from '../../components/StockCard';

describe('StockCard component integrity', () => { 
  it('renders correctly', () => {
    const navbar = renderer.create(
      <BrowserRouter>
        <StockCard />
      </BrowserRouter>,
    ).toJSON();
    expect(navbar).toMatchSnapshot();
  });
});
