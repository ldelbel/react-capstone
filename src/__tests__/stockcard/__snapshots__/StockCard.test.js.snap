// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StockCard DOM renders correctly 1`] = `
<div
  className="sc-bdfBwQ tSdqh"
>
  <div
    className="d-flex flex-column justify-content-start align-items-center"
    style={
      Object {
        "height": "75%",
      }
    }
  >
    <div
      className="sc-gsTCUz hgATcl d-flex flex-row align-items-center"
    >
      <div
        className="d-flex align-items-center"
        style={
          Object {
            "height": "100%",
            "width": "50%",
          }
        }
      >
        <div
          className="sc-dlfnbm eKBrhu"
          style={
            Object {
              "backgroundImage": "url(https://financialmodelingprep.com/image-stock/AAPL.jpg)",
            }
          }
        />
      </div>
      <div
        className="d-flex flex-column justify-content-start align-items-end"
        style={
          Object {
            "height": "100%",
            "width": "50%",
          }
        }
      >
        <span
          style={
            Object {
              "color": "var(--dark)",
              "fontFamily": "Alfa Slab One, cursive",
              "fontSize": "25px",
              "height": "30px",
            }
          }
        >
          AAPL
        </span>
        <span
          style={
            Object {
              "color": "var(--blue)",
              "fontFamily": "Alfa Slab One, cursive",
              "fontSize": "10px",
            }
          }
        >
          NASDAQ
        </span>
      </div>
    </div>
    <div
      style={
        Object {
          "height": "55%",
          "padding": "0",
          "width": "100%",
        }
      }
    >
      <div
        className="d-flex align-items-end"
        style={
          Object {
            "height": "40%",
            "paddingLeft": "10px",
          }
        }
      >
        <span
          className="text-nowrap text-truncate flex-nowrap"
          style={
            Object {
              "fontFamily": "Alata, sans-serif",
              "fontSize": "18px",
            }
          }
        >
          Apple Inc
        </span>
      </div>
      <div
        className="d-flex flex-column justify-content-center align-items-end"
        style={
          Object {
            "fontSize": "13px",
            "height": "60%",
          }
        }
      >
        <div
          className="sc-hKgILt fXVWRg"
        >
          <span>
            Technology
          </span>
        </div>
        <div
          className="sc-eCssSg iVXGEU"
        >
          <span>
            Consumer Electronics
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="sc-jSgupP foFzOL d-flex flex-row justify-content-around align-items-center"
  >
    <button
      className="sc-gKsewC jbnqRc btn btn-primary"
      onClick={[Function]}
      type="button"
    >
      Details
    </button>
    <span
      style={
        Object {
          "color": "var(--yellow)",
          "fontFamily": "Alfa Slab One, cursive",
          "fontSize": "23px",
        }
      }
    >
      U$ 135.37
    </span>
  </div>
</div>
`;
